src/wallet/wallet.h

```diff
@@ -1218,4 +1222,37 @@ bool CWallet::DummySignTx(CMutableTransaction &txNew, const ContainerType &coins
     return true;
 }
 
+/** RAII object to check and reserve a wallet rescan */
+class WalletRescanReserver
+{
+private:
+    CWalletRef m_wallet;
+    bool m_could_reserve;
+public:
+    explicit WalletRescanReserver(CWalletRef w) : m_wallet(w), m_could_reserve(false) {}
+
+    bool reserve()
+    {
+        std::lock_guard<std::mutex> lock(m_wallet->mutexScanning);
+        if (m_wallet->fScanningWallet) {
+            return false;
+        }
+        m_wallet->fScanningWallet = true;
+        m_could_reserve = true;
+        return true;
+    }
+
+    bool isReserved() const {
+        return (m_could_reserve && m_wallet->fScanningWallet);
```

Comment:
`return m_could_reserve` is enough? `WalletRescanReserver` is the "owner.

Reply:
I think it's better for future changes to keep the check here... don't hurt but may protect from a future mistake.

Reply:
In commit "Make sure WalletRescanReserver has successfully reserved".

Alternative could be to just `return m_could_reserve` and assert `!m_could_reserve || fScanningWallet`, so if there is a future mistake, it can be detected instead of papered over.

Also formatting here is inconsistent, opening brace for function goes on new line.